5813 show geolocation on osm

zaziemo opened the pull request:
Things @Realtin and I still have to work on but where we need discussion/decision or help:

**Tile Server**
Currently we are using mapbox.com which provides free access for up to 50.000 requests per month
- [x]  should every podmin get their own key or should we negotiate one access for everyone
- [x] should we stick to using mapbox or should we probably use another tile server  
- [x] how and where to store the tileserver configuration into the AppConfig?

**Tests**
We implemented some Jasmine-Tests but one of them is not working: https://github.com/TeamDeltaQuadrant/diaspora/blob/5813-show-geolocation-on-osm/spec/javascripts/app/views/location_stream_spec.js (see comment)

**View/Subview**
~~Currently we add the map container div with jquery which is probably not the most elegant solution. Should we add a subview for the map and if so: How do we do that?~~

zaziemo committed:
add a map subview

- add coordinates in post_presenter
- add map to the SPV if location is provided
- add leaflet.js to render map and marker of position
- make coordinates available in frontend
- add map scss
- make stream post location clickable and redirect to the SPV
- prevent render map if no location data is provided
- add tests for coordinates
- use the leaflet gem instead of the JS assets
(#5813)

Flaburgan commented:
Awesome! Can we have a screenshot of how it looks?

Realtin commented:
![bildschirmfoto 2015-07-28 um 15 57 00](https://cloud.githubusercontent.com/assets/507997/8933133/7f72cacc-3541-11e5-99f8-36f95264d1e3.png)

jaywink commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -37,6 +37,42 @@ app.views.SinglePostContent = app.views.Base.extend({
+  },
+
+  postRenderTemplate : function(){
+    _.defer(_.bind(this.map, this));
```
Awesome start! :) I guess this means the map will open up automatically on loading the post? How do we feel about that, I'm thinking it might be nice to show it on request after for example clicking the location text. Then it could work also in the stream?

Flaburgan replied:
If it's automatically loaded, I think I want it with a smaller height. In any case, it should be displayed before the polls (where the other users interact), maybe even inside `#head`

zaziemo replied:
@jaywink yes, the map will open up automatically in the single post view. In the stream it is not shown automatically but a click on the location leads to the single post view.
@Flaburgan it is no problem to have it placed before the polls or make it smaller. Inside the head it could at most have a thumbnail size … but we can try that as well.

svbergerem replied:
I think loading the map in the SPV automatically is no problem. In the stream I don't think we should load the map automatically and this PR also doesn't do that currently. Moving the map to a subview could help us to display the map in the stream on click without going to the SPV.

UI wise there is currently the problem that the location text and the map are not next to each other. We could move the map to the `#head` section of the SPV but I am not sure about that. In the stream we add the location at the end of the post. We could do the same thing here and then it would be next to the map. @Flaburgan what do you think about that?

jhass commented on spec/models/status_message_spec.rb:
```diff
@@ -112,105 +107,104 @@
 
-        expect{
-          @sm.create_mentions
+        expect {
```
Ignore these, Hound is ignoring our config here

jhass commented on app/presenters/post_presenter.rb:
```diff
@@ -19,36 +19,37 @@ def as_json(options={})
+        likes:          [user_like].compact,
+        reshares:       [user_reshare].compact,
+        comments_count: @post.comments_count,
+        likes_count:    @post.likes_count,
```
Again you used the correct style here.

jhass commented:
https://rubygems.org/gems/leaflet-rails seems to be fairly popular, please try using it instead of copying the leaflet assets into the project.

jhass commented on app/assets/templates/status-message-location_tpl.jst.hbs:
```diff
@@ -1,5 +1,5 @@
 {{#if location}}
   <div class='near-from'>
-    {{ t "publisher.near_from" location=location}}
+    <a href="/posts/{{id}}">{{ t "publisher.near_from" location=location}}</a>
```
Not sure this should be an actual link, if anything then to some external map showing the POI. But I'm leaning more towards just styling it as one with `cursor: pointer` and `text-decoration: underline` on the hover state of the parent container.

zaziemo replied:
@jhass I'm not sure if I get that right. Do you want us to style the location in the stream with `cursor: pointer` and `text-decoration: underline` that should get visible via hovering? What about the link to the single post view (that's where the map is right now)? Should it stay?

jhass replied:
I didn't realize that you're linking to the SPV, I thought you'd already pop-in the map when clicking on the location.

jhass commented on app/presenters/post_presenter.rb:
```diff
@@ -19,36 +19,37 @@ def as_json(options={})
+      root:                         root,
+      title:                        title,
+      address:                      @post.address,
+      coordinates:                  @post.coordinates,
```
I think it might make sense to group these under a `location` key:

```rb
location: {address: @post.location.address, coordinates: @post.location.coordinates}
```

zaziemo replied:
@jhass we thought of that as well. But we wanted to change it already in the model. We postponed it because of all the refactoring :)

Realtin commented:
@jhass We first tried to use the gem but we were not sure how we could use for example:
```
<%= map(:center => {
  :latlng => [51.52238797921441, -0.08366235665359283],
  :zoom => 18
}) %>
```
(_from the [gem documentation](https://github.com/axyjo/leaflet-rails)_)

in the javascript views. 
Is there a way to do that?

jhass commented:
Not really no, but you can still use the assets from the gem and then doing it like you do it now should work too.

svbergerem commented:
> - should every podmin get their own key or should we negotiate one access for everyone
> - should we stick to using mapbox or should we probably use another tile server

I already used mapbox for some other projects and AFAIR they had free keys for open source projects (with more than 50000 tiles/month) but I wasn't able to find any information right now on their website.

There are also some other free tile servers listed in the [OSM wiki](https://wiki.openstreetmap.org/wiki/Tile_usage_policy#Alternative_OpenStreetMap_Tile_Providers), maybe we could use [MapQuest](https://wiki.openstreetmap.org/wiki/MapQuest#MapQuest-hosted_map_tiles)?

svbergerem commented:
> Currently we add the map container div with jquery which is probably not the most elegant solution. Should we add a subview for the map and if so: How do we do that?

Currently you only add this one div but we might want to use the map also somewhere else and adding the code for leaflet every time sounds bad. A subview sounds like a better solution. I think you can compare this subview to the subview for the location. There is [a template](https://github.com/diaspora/diaspora/blob/develop/app/assets/templates/status-message-location_tpl.jst.hbs), [a view](https://github.com/diaspora/diaspora/blob/develop/app/assets/javascripts/app/views/location_stream.js) (you could add a `initialize` function in the new map view where you add the leaflet code) and it is [defined as a subview](https://github.com/diaspora/diaspora/blob/develop/app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js#L13) (the first string is the selector for the element where you want to add your subview, the second one is the function name to add the subview). To make the map view as generic as possible maybe accept the coordinates as an options hash in the initialize function. (Like in the [tags view](https://github.com/diaspora/diaspora/blob/develop/app/assets/javascripts/app/views/tags_view.js#L4))

denschub commented:
I'll contact Mapbox in the name of the project today and ask for an OpenSource license.

zaziemo commented:
@denschub yay! \o/

zaziemo commented:
@svbergerem we now outsourced the map in a subview. Thanks for the detailed information :)

jhass commented on db/schema.rb:
```diff
@@ -507,73 +510,73 @@
+  add_index "users", ["username"], name: "index_users_on_username", unique: true, using: :btree
 
   add_foreign_key "aspect_memberships", "aspects", name: "aspect_memberships_aspect_id_fk", on_delete: :cascade
   add_foreign_key "aspect_memberships", "contacts", name: "aspect_memberships_contact_id_fk", on_delete: :cascade
```
Since you didn't add any migration, please try not to commit any changes to `db/schema.rb`.

Realtin replied:
I have no idea where that comes from. we didn't do anything in the schema .. 
maybe it happen while rebasing somehow. :/

zaziemo replied:
@jhass can we revert that? I'm not even sure how that change came in. We didn't delete `length: {"username"=>191}` probably this came through rebasing/merging?

jhass replied:
Since it's the last commit, we can fairly easily edit it:

```
git checkout upstream/develop -- db/schema.rb
git commit --amend db/schema.rb
git push -f origin 5813-show-geolocation-on-osm
```

This will replace the last commit, so the other one of you will have to `git fetch` and `git reset --hard origin/5813-show-geolocation-on-osm` their branch, like after a rebase by somebody else.

svbergerem replied:
`db/schema.rb` sometimes changes when you run migrations. Just make sure that you don't add those changes to your commit. (be careful with `commit -a`)

jhass replied:
Another good tip is to always use `git add -p` and `git commit -p`, which makes you review all changes you're going to add/commit.

Realtin replied:
@svbergerem that was probably it! @zaziemo mysql exploded when the cucumber tests crashed her computer. so she changed to postgres 

we will be more careful from now on!

jhass commented:
To actually answer on how to add configuration and how to get it into the frontend:

You can introduce new settings by providing (possibly empty) default values in `config/defaults.yml`, and adding commented examples in `config/diaspora.yml.example`. These will then become available as `AppConfig.key.subkey.subsubkey`. You can then pass them along to the frontend [here](https://github.com/diaspora/diaspora/blob/develop/app/controllers/application_controller.rb#L142-L147) and access them via `gon.appConfig.foo.bar` in the JS.

And thinking about it please do so, if @denschub has luck in obtaining an API key we can redistribute, just skip the step of adding examples in `config/diaspora.yml.example` and add it as default.

zaziemo committed:
merge address & coordinates in one location object

to provide a clear arrangement of all location data
- add 'L' to predefs for pronto because it is part of the leaflet library
- fix: show address template only with an address present
with merging the location objects into one, only the objects within
the location object can be empty
(#5813)

diaspora-code-review commented on app/assets/javascripts/app/views/location_map.js:
```diff
@@ -0,0 +1,34 @@
+          attribution:  attribution,
+          maxZoom: 18,
+          id: "zaziemo.mpn66kn8",
+          accessToken: "pk.eyJ1IjoiemF6aWVtbyIsImEiOiI3ODVjMzVjNmM2ZTU3YWM3YTE5YWYwMTRhODljM2M1MSJ9.-nVgyS4PLnV4m9YkvMB5wA"
```
Line is too long.

diaspora-code-review commented on app/assets/javascripts/app/views/location_map.js:
```diff
@@ -0,0 +1,34 @@
+      // if (coordinates != "" && tileserver.enable) {  // for when the tileserver is set via the diaspora.yml
+      if (coordinates.lat) {
+        var tileLayerSource = "https://api.tiles.mapbox.com/v4/{id}/{z}/{x}/{y}.png?access_token={accessToken}";
+        var map = L.map("map").setView([coordinates.lat, coordinates.lng], 16);
```
'L' is not defined.

diaspora-code-review replied:
'L' is not defined.

diaspora-code-review commented on app/assets/javascripts/app/views/location_map.js:
```diff
@@ -0,0 +1,34 @@
+                          "<a href='http://creativecommons.org/licenses/by-sa/2.0/''>CC-BY-SA</a>, " +
+                          "Imagery © <a href='http://mapbox.com'>Mapbox</a>";
+
+        L.tileLayer(tileLayerSource, {
```
'L' is not defined.

jhass replied:
You might want to add this to the predefs in `.jshintrc`

diaspora-code-review commented on app/assets/javascripts/app/views/location_map.js:
```diff
@@ -0,0 +1,34 @@
+          accessToken: "pk.eyJ1IjoiemF6aWVtbyIsImEiOiI3ODVjMzVjNmM2ZTU3YWM3YTE5YWYwMTRhODljM2M1MSJ9.-nVgyS4PLnV4m9YkvMB5wA"
+        }).addTo(map);
+
+        var markerOnMap = L.marker(coordinates).addTo(map);
```
'L' is not defined.

diaspora-code-review replied:
'markerOnMap' is defined but never used.

diaspora-code-review replied:
'L' is not defined.

diaspora-code-review replied:
'markerOnMap' is defined but never used.

diaspora-code-review commented on app/presenters/post_presenter.rb:
```diff
@@ -19,36 +19,37 @@ def as_json(options={})
+      address:                      @post.address,
+      coordinates:                  @post.coordinates,
+      poll:                         @post.poll(),
+      already_participated_in_poll: already_participated_in_poll,
```
Do not use parentheses for method calls with no arguments.

diaspora-code-review replied:
Do not use parentheses for method calls with no arguments.

diaspora-code-review commented on app/assets/javascripts/app/views/location_map.js:
```diff
@@ -0,0 +1,34 @@
+                          "<a href='http://creativecommons.org/licenses/by-sa/2.0/''>CC-BY-SA</a>, " +
+                          "Imagery © <a href='http://mapbox.com'>Mapbox</a>";
+
+        L.tileLayer(tileLayerSource, {
```
Line is too long.

diaspora-code-review commented on app/assets/javascripts/app/views/location_map.js:
```diff
@@ -0,0 +1,34 @@
+          attribution:  attribution,
+          maxZoom: 18,
+          id: "zaziemo.mpn66kn8",
+          accessToken: "pk.eyJ1IjoiemF6aWVtbyIsImEiOiI3ODVjMzVjNmM2ZTU3YWM3YTE5YWYwMTRhODljM2M1MSJ9.-nVgyS4PLnV4m9YkvMB5wA"
```
'markerOnMap' is defined but never used.

diaspora-code-review commented on app/presenters/post_presenter.rb:
```diff
@@ -19,36 +19,37 @@ def as_json(options={})
+      nsfw:                         @post.nsfw,
+      author:                       @post.author.as_api_response(:backbone),
+      o_embed_cache:                @post.o_embed_cache.try(:as_api_response, :backbone),
+      open_graph_cache:             @post.open_graph_cache.try(:as_api_response, :backbone),
```
Line is too long. [89/80]

svbergerem replied:
Our limit is 120 chars so the review bot is obviously not using our config.

@zaziemo @Realtin as long as the bot is using the wrong config you can ignore these comments and test the code style locally with pronto.

diaspora-code-review commented on app/presenters/post_presenter.rb:
```diff
@@ -19,36 +19,37 @@ def as_json(options={})
+      author:                       @post.author.as_api_response(:backbone),
+      o_embed_cache:                @post.o_embed_cache.try(:as_api_response, :backbone),
+      open_graph_cache:             @post.open_graph_cache.try(:as_api_response, :backbone),
+      mentioned_people:             @post.mentioned_people.as_api_response(:backbone),
```
Line is too long. [92/80]

diaspora-code-review commented on app/presenters/post_presenter.rb:
```diff
@@ -19,36 +19,37 @@ def as_json(options={})
+      o_embed_cache:                @post.o_embed_cache.try(:as_api_response, :backbone),
+      open_graph_cache:             @post.open_graph_cache.try(:as_api_response, :backbone),
+      mentioned_people:             @post.mentioned_people.as_api_response(:backbone),
+      photos:                       @post.photos.map {|p| p.as_api_response(:backbone) },
```
Line is too long. [86/80]

diaspora-code-review commented on app/presenters/post_presenter.rb:
```diff
@@ -19,36 +19,36 @@ def as_json(options={})
       @post.raw_message
```
Line is too long. [89/80]

diaspora-code-review replied:
Space between { and | missing.

diaspora-code-review commented on app/presenters/post_presenter.rb:
```diff
@@ -19,36 +19,36 @@ def as_json(options={})
-        :id => @post.id,
-        :guid => @post.guid,
-        :text => text,
-        :public => @post.public,
```
Do not use parentheses for method calls with no arguments.

Realtin committed:
add map to header

and add toggle map function
to show and load map only when user is clicking on address

(#5813)

zaziemo commented:
Hey there,
we tried a different positioning of the map:
* in header
* only loads and opens if requested
* works the same in the single post view and the stream

We recorded a small webcast for you to watch: https://vimeo.com/135673364
You'll find the code here: https://github.com/TeamDeltaQuadrant/diaspora/tree/5813-add-map-to-header

What do you think? Should we merge that into this pull request (after having implemented the translation and toggling of the text)?

@denschub do you have news from mapbox about a possible unlimited use of their tiles?

Flaburgan commented:
>UI wise there is currently the problem that the location text and the map are not next to each other. We could move the map to the #head section of the SPV but I am not sure about that. In the stream we add the location at the end of the post. We could do the same thing here and then it would be next to the map. @Flaburgan what do you think about that?

I think the map and the location are meta info about the post and should be in the head in the SPV, next to the other meta info (visibility, date...). A suggestion: always display the map in the SPV, with a 100% width but a really small height (50px?) and expand the height when the user click on it:
![map](https://cloud.githubusercontent.com/assets/930064/9149738/d3f9612c-3db7-11e5-9d3e-0569af010fe3.png)

I don't know about the stream. I just noticed a bug on your video though, the address (and the map) should be inside the post when reshared, not on the reshare itself (notice the left border). This is not related to the map though.

![capture du 2015-08-08 10 17 58](https://cloud.githubusercontent.com/assets/930064/9149744/04a50ee8-3db8-11e5-8adf-959b18856731.png)

denschub commented:
@zaziemo Sitll in detailed discussions but now it's weekend, so we have to be patient a bit :)

zaziemo commented:
@Flaburgan thanks for your suggestions. We liked the idea of only showing the map after clicking on "open map" because only then the tiles from the tileserver are loaded (which might be interesting for performance and capacity reasons). But it's also nice to have a visual representation of the map whenever you open the single post view.

Concerning the reshare bug: we can have a look at this too.

zaziemo commented:
@denschub ok, we are very oooooommmm :)

zaziemo committed:
add a different map representation to the SPV

so a small map is rendered in the SPV and a location is provided
via toggling the map gets bigger

(#5813)

Realtin committed:
rename location_view to locator

because it is confusing with the other location files

zaziemo commented on spec/javascripts/app/views/location_stream_spec.js:
```diff
@@ -0,0 +1,39 @@
+        expect(this.view.$el.find(".mapContainer")[0]).not.toHaveClass("empty");
+      });
+      /*
+       * does not work .. not sure why
```
we have problems to get that test running. The function call toggleMap seems not do do anything inside this particular it-block.

svbergerem replied:
The function call initializes the map like in the test above. The `display: block` to `display: none` change only happens after the map has been initialized so you will have to initialize the map first by calling `this.view.toggleMap()` at the beginning of your test and only test the behaviour of the second call.

denschub commented:
So, apparently we are not able to get a Access Token for multiple domains. The project team would be required to create new tokens for each pod that is interested in using the Mabbox tiles. So we could either ask podmins to send a mail to team@diasporafoundation.org or to create their own mapbox-account. What do others think?

zaziemo committed:
outsource map credentials

and allow podmin to enable and disable the feature
(#5813)

Realtin commented:
Ok @christophe-de pointed me to the [OpenMapSurfer](http://korona.geog.uni-heidelberg.de) .. it is a free TileServer at Uni Heidelberg for non commercial use and no accounts!. But we have to contact them.

It looks like this:

![bildschirmfoto 2015-08-22 um 15 56 09](https://cloud.githubusercontent.com/assets/507997/9424270/c65639a2-48e6-11e5-9adb-ed23cac7ade5.png)

![bildschirmfoto 2015-08-22 um 15 59 46](https://cloud.githubusercontent.com/assets/507997/9424275/dbea5b4a-48e6-11e5-948b-beb5a84ee567.png)

jaywink commented:
Would it be impossible to make the map tiles location configurable, by specifying `source` and optional `username` and `token`, if required by the source? Then we can have multiple options available for podmins to choose from. I'm afraid if we choose a single one, for example the Uni Heidelberg, if it performs badly or goes down it affects everybody.

This way we could also have MapBox as one option but say in the configuration that this source needs an account.

Realtin committed:
set maptile default to OpenMapSurfer

- refactor code for toggle map function
- adjust jasmine tests for maps and add tests for SPV
- change name of access token in defaults.yml according to naming conventions
- add explanation of map use for podmins
- add location to post_presenter
- fix opening multiple maps inside the stream when clicking on a reshare location
(#5813)

diaspora-code-review commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,51 @@
+        if (location.lat) {
+          // If the mapbox option is enabled in the defaults the mapbox tiles with the podmin's credentials are used.
+          // If mapbox is not enabled the OpenMapSurfer tiles are used, which don't need credentials.
+          var mapsource = gon.appConfig.map.mapbox.enabled ? gon.appConfig.map.mapbox : "";
```
Line is too long.

diaspora-code-review commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,51 @@
+          var mapsource = gon.appConfig.map.mapbox.enabled ? gon.appConfig.map.mapbox : "";
+          var tileLayerSource = mapsource ? "https://api.tiles.mapbox.com/v4/{id}/{z}/{x}/{y}.png?access_token={accessToken}"
+                                          : "http://korona.geog.uni-heidelberg.de/tiles/roads/x={x}&y={y}&z={z}";
+          var tileAttribution = mapsource ? "<a href='https://www.mapbox.com'>Mapbox</a>"
```
Line is too long.

diaspora-code-review commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,51 @@
+          var attribution = "Map data &copy; <a href='http://openstreetmap.org'>OpenStreetMap</a> contributors, " +
+                            "<a href='http://creativecommons.org/licenses/by-sa/2.0/''>CC-BY-SA</a>, " +
+                            "Imagery © "+ tileAttribution;
+
```
'L' is not defined.

diaspora-code-review commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,51 @@
+                            "<a href='http://creativecommons.org/licenses/by-sa/2.0/''>CC-BY-SA</a>, " +
+                            "Imagery © "+ tileAttribution;
+
+          var map = L.map(mapContainer[0]).setView([location.lat, location.lng], 14);
```
'L' is not defined.

diaspora-code-review commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,51 @@
+            attribution:  attribution,
+            maxZoom: 18,
+          }).addTo(map);
+
```
'L' is not defined.

diaspora-code-review replied:
'markerOnMap' is defined but never used.

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -23,8 +26,56 @@ app.views.SinglePostContent = app.views.Base.extend({
+                                      : "http://korona.geog.uni-heidelberg.de/tiles/roads/x={x}&y={y}&z={z}";
+      var tileAttribution = mapsource ? "<a href='https://www.mapbox.com'>Mapbox</a>"
+                                      : "<a href='http://korona.geog.uni-heidelberg.de/contact.html'>OpenMapSurfer</a>";
+      var attribution = "Map data &copy; <a href='http://openstreetmap.org'>OpenStreetMap</a> contributors, " +
```
Line is too long.

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -23,8 +26,49 @@ app.views.SinglePostContent = app.views.Base.extend({
+      } else {
+          this.$el.find(".mapContainer").css("height", "75px");
+          this.$el.find(".leaflet-control-zoom").css("display", "none");
+      }
```
Unnecessary semicolon.

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -23,8 +26,56 @@ app.views.SinglePostContent = app.views.Base.extend({
+      L.tileLayer(tileLayerSource, {
+        id: mapsource.id,
+        accessToken: mapsource.accessToken,
+        attribution:  attribution,
```
'L' is not defined.

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -23,8 +26,56 @@ app.views.SinglePostContent = app.views.Base.extend({
+        id: mapsource.id,
+        accessToken: mapsource.accessToken,
+        attribution:  attribution,
+        maxZoom: 18,
```
'L' is not defined.

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -23,8 +26,49 @@ app.views.SinglePostContent = app.views.Base.extend({
+        this.$el.find(".mapContainer").css("height", "200px");
+        this.$el.find(".leaflet-control-zoom").css("display", "block");
+      } else {
+          this.$el.find(".mapContainer").css("height", "75px");
```
'L' is not defined.

diaspora-code-review replied:
'markerOnMap' is defined but never used.

diaspora-code-review commented on spec/javascripts/app/views/location_stream_spec.js:
```diff
@@ -0,0 +1,39 @@
+      it("should not initialize the map", function() {
+        expect(this.view.$el[0]).not.toContainElement(".mapContainer");
+      });
+    });
```
Expected an assignment or function call and instead saw an expression.

diaspora-code-review replied:
Expected an assignment or function call and instead saw an expression.

diaspora-code-review replied:
Expected an assignment or function call and instead saw an expression.

diaspora-code-review commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,51 @@
+    "click .near-from": "toggleMap"
+  },
+  templateName: "status-message-location",
+
```
Line is too long.

diaspora-code-review commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,51 @@
+
+  toggleMap: function () {
+    if (gon.appConfig.map.enabled){
+      var mapContainer = this.$el.find(".mapContainer");
```
Line is too long.

diaspora-code-review commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,51 @@
+                                          : "<a href='http://korona.geog.uni-heidelberg.de/contact.html'>OpenMapSurfer</a>";
+          var attribution = "Map data &copy; <a href='http://openstreetmap.org'>OpenStreetMap</a> contributors, " +
+                            "<a href='http://creativecommons.org/licenses/by-sa/2.0/''>CC-BY-SA</a>, " +
+                            "Imagery © "+ tileAttribution;
```
'markerOnMap' is defined but never used.

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -1,6 +1,10 @@
+
   templateName: 'single-post-viewer/single-post-content',
   tooltipSelector: "time, .post_scope",
 
```
Line is too long.

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -23,8 +26,49 @@ app.views.SinglePostContent = app.views.Base.extend({
+      var tileLayerSource = "https://api.tiles.mapbox.com/v4/{id}/{z}/{x}/{y}.png?access_token={accessToken}";
+      var map = L.map(mapContainer[0]).setView([location.lat, location.lng], 14);
+      var attribution = "Map data &copy; <a href='http://openstreetmap.org'>OpenStreetMap</a> contributors, " +
+                        "<a href='http://creativecommons.org/licenses/by-sa/2.0/''>CC-BY-SA</a>, " +
```
Unnecessary semicolon.

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -23,8 +26,49 @@ app.views.SinglePostContent = app.views.Base.extend({
+      // get location data and render map
+      var location = this.model.get("location");
+      var tileLayerSource = "https://api.tiles.mapbox.com/v4/{id}/{z}/{x}/{y}.png?access_token={accessToken}";
+      var map = L.map(mapContainer[0]).setView([location.lat, location.lng], 14);
```
'markerOnMap' is defined but never used.

diaspora-code-review commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,51 @@
+          // If the mapbox option is enabled in the defaults the mapbox tiles with the podmin's credentials are used.
+          // If mapbox is not enabled the OpenMapSurfer tiles are used, which don't need credentials.
+          var mapsource = gon.appConfig.map.mapbox.enabled ? gon.appConfig.map.mapbox : "";
+          var tileLayerSource = mapsource ? "https://api.tiles.mapbox.com/v4/{id}/{z}/{x}/{y}.png?access_token={accessToken}"
```
Line is too long.

diaspora-code-review commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,51 @@
+          var tileLayerSource = mapsource ? "https://api.tiles.mapbox.com/v4/{id}/{z}/{x}/{y}.png?access_token={accessToken}"
+                                          : "http://korona.geog.uni-heidelberg.de/tiles/roads/x={x}&y={y}&z={z}";
+          var tileAttribution = mapsource ? "<a href='https://www.mapbox.com'>Mapbox</a>"
+                                          : "<a href='http://korona.geog.uni-heidelberg.de/contact.html'>OpenMapSurfer</a>";
```
Line is too long.

diaspora-code-review commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,51 @@
+            maxZoom: 18,
+          }).addTo(map);
+
+          var markerOnMap = L.marker(location).addTo(map);
```
'markerOnMap' is defined but never used.

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -23,8 +26,56 @@ app.views.SinglePostContent = app.views.Base.extend({
+      // If the mapbox option is enabled in the defaults the mapbox tiles with the podmin's credentials are used.
+      // If mapbox is not enabled the OpenMapSurfer tiles are used, which don't need credentials.
+      var mapsource = gon.appConfig.map.mapbox.enabled ? gon.appConfig.map.mapbox : "";
+      var tileLayerSource = mapsource ? "https://api.tiles.mapbox.com/v4/{id}/{z}/{x}/{y}.png?access_token={accessToken}"
```
Line is too long.

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -23,8 +26,56 @@ app.views.SinglePostContent = app.views.Base.extend({
+      // put marker on map
+      var markerOnMap = L.marker(location).addTo(map);
+      return map;
+      };
```
Unnecessary semicolon.

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -23,8 +26,56 @@ app.views.SinglePostContent = app.views.Base.extend({
+      map.invalidateSize();
+
+      // put marker on map
+      var markerOnMap = L.marker(location).addTo(map);
```
'markerOnMap' is defined but never used.

diaspora-code-review commented on spec/javascripts/helpers/factory.js:
```diff
@@ -148,6 +148,14 @@ var factory = {
 
+  location : function() {
+    return {
+      address: "Starco Mart, Mission Street, West SoMa, San Francisco, San Francisco City and County, Kalifornien, 94103, Vereinigte Staaten von Amerika",
```
Line is too long.

diaspora-code-review commented on app/presenters/post_presenter.rb:
```diff
@@ -19,36 +19,36 @@ def as_json(options={})
+      root:                         root,
+      title:                        title,
+      location:                     @post.post_location,
+      poll:                         @post.poll(),
```
Do not use parentheses for method calls with no arguments.

diaspora-code-review commented on spec/factories.rb:
```diff
@@ -140,9 +140,10 @@ def r_str
+    address "Starco Mart, Mission Street, West SoMa, San Francisco, San Francisco "\
+            "City and County, Kalifornien, 94103, Vereinigte Staaten von Amerika"
+    lat 37.78
+    lng -122.41
```
Ambiguous negative number operator. Parenthesize the method arguments if it's surely a negative number operator, or add a whitespace to the right of the `-` if it should be a subtraction.

jhass replied:
Ping.

diaspora-code-review commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,47 @@
+                            "<a href='http://creativecommons.org/licenses/by-sa/2.0/''>CC-BY-SA</a>, " +
+                            "Imagery © "+ tileAttribution;
+
+          var map = L.map(mapContainer[0]).setView([location.lat, location.lng], 14);
```
'L' is not defined.

diaspora-code-review commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,47 @@
+                            "Imagery © "+ tileAttribution;
+
+          var map = L.map(mapContainer[0]).setView([location.lat, location.lng], 14);
+          L.tileLayer(tileLayerSource, {
```
'L' is not defined.

diaspora-code-review commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,47 @@
+            maxZoom: 18,
+          }).addTo(map);
+
+          var markerOnMap = L.marker(location).addTo(map);
```
'L' is not defined.

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -23,8 +26,52 @@ app.views.SinglePostContent = app.views.Base.extend({
+
+  toggleMap: function () {
+    if (gon.appConfig.map.enabled){
+      $('.mapContainer').height($('.small-map')[0] ? 200 : 50);
```
Strings must use doublequote.

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -23,8 +26,52 @@ app.views.SinglePostContent = app.views.Base.extend({
+  toggleMap: function () {
+    if (gon.appConfig.map.enabled){
+      $('.mapContainer').height($('.small-map')[0] ? 200 : 50);
+      $('.leaflet-control-zoom').css("display", $('.small-map')[0] ? "block" : "none");
```
Strings must use doublequote.

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -23,8 +26,52 @@ app.views.SinglePostContent = app.views.Base.extend({
+                        "<a href='http://creativecommons.org/licenses/by-sa/2.0/''>CC-BY-SA</a>, " +
+                        "Imagery © "+ tileAttribution;
+
+      var map = L.map(mapContainer[0]).setView([location.lat, location.lng], 14);
```
'L' is not defined.

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -23,8 +26,52 @@ app.views.SinglePostContent = app.views.Base.extend({
+                        "Imagery © "+ tileAttribution;
+
+      var map = L.map(mapContainer[0]).setView([location.lat, location.lng], 14);
+      L.tileLayer(tileLayerSource, {
```
'L' is not defined.

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -23,8 +26,52 @@ app.views.SinglePostContent = app.views.Base.extend({
+      map.invalidateSize();
+
+      // put marker on map
+      var markerOnMap = L.marker(location).addTo(map);
```
'L' is not defined.

diaspora-code-review commented on spec/javascripts/app/views/location_stream_spec.js:
```diff
@@ -0,0 +1,39 @@
+    context("with location provided", function() {
+      beforeEach(function(){
+        this.post.set({location : factory.location()}); // set location
+        gon.appConfig = { map: {enabled: true, mapbox: {enabled: true, id: "yourID", accessToken: "yourAccessToken" }} };
```
Line is too long.

zaziemo commented on spec/javascripts/app/views/location_stream_spec.js:
```diff
@@ -0,0 +1,39 @@
+        expect(this.view.$el.find(".mapContainer")[0]).not.toHaveClass("empty");
+      });
+
+      it("should change display status on every click", function() {
```
@svbergerem we still cannot get this test passing :(  Any more ideas?

svbergerem replied:
It looks like the jQuery function `toggle` only works if the element has been added to the pages html, so I added the rendered html to `spec.content()` which is element where you want to add your html inside of a jasmine test. I also renamed the spec to `app.views.LocationStream`, since this is the name of the view you are testing, and unified the tests afterwards. Here is the diff:
```diff
diff --git a/spec/javascripts/app/views/location_stream_spec.js b/spec/javascripts/app/views/location_stream_spec.js
index 09186bc..055541c 100644
--- a/spec/javascripts/app/views/location_stream_spec.js
+++ b/spec/javascripts/app/views/location_stream_spec.js
@@ -1,38 +1,43 @@
-describe("app.views.LocationMap", function() {
+describe("app.views.LocationStream", function() {
   beforeEach(function(){
     this.post = factory.post();
     this.view = new app.views.LocationStream({model : this.post});
+    gon.appConfig = { map: {enabled: true, mapbox: {enabled: true, id: "yourID", accessToken: "yourAccessToken" }} };
   });
 
   describe("toggleMap", function() {
     context("with location provided", function() {
       beforeEach(function(){
         this.post.set({location : factory.location()}); // set location
-        gon.appConfig = { map: {enabled: true, mapbox: {enabled: true, id: "yourID", accessToken: "yourAccessToken" }} };
-        this.view.render();
+        spec.content().html(this.view.render().el);
       });
 
       it("should contain a map container", function() {
-        expect(this.view.$el[0]).toContainElement(".mapContainer");
+        expect(spec.content()).toContainElement(".mapContainer");
       });
 
       it("should initialize map", function() {
-        expect(this.view.$el.find(".mapContainer")[0]).toHaveClass("empty");
+        expect($(".mapContainer")).toHaveClass("empty");
         this.view.toggleMap();
-        expect(this.view.$el.find(".mapContainer")[0]).not.toHaveClass("empty");
+        expect($(".mapContainer")).not.toHaveClass("empty");
       });
 
       it("should change display status on every click", function() {
         this.view.toggleMap();
-        expect(this.view.$el.find(".mapContainer")[0]).toHaveCss({display: "block"});
+        expect($(".mapContainer")).toHaveCss({display: "block"});
         this.view.toggleMap();
-        expect(this.view.$el.find(".mapContainer")[0]).toHaveCss({display: "none"});
+        expect($(".mapContainer")).toHaveCss({display: "none"});
       });
 
-    }),
+    });
+
     context("without location provided", function() {
+      beforeEach(function(){
+        spec.content().html(this.view.render().el);
+      });
+
       it("should not initialize the map", function() {
-        expect(this.view.$el[0]).not.toContainElement(".mapContainer");
+        expect(spec.content()).not.toContainElement(".mapContainer");
       });
     });
   });
```

zaziemo replied:
Wow, @svbergerem thanks for figuring that out :) 
Btw, how did you find that solution? Can we somewhere read anything about that spec.content construction?

svbergerem replied:
> Btw, how did you find that solution?

First I found out (with some `console.log(...)`) that the toggle function really has been called but didn't do anything. Then I called the toggle function manually in the spec and still saw no change so I wanted to get a minimal failing spec and added some div to `spec.content()` and called `toggle()` on that div but that worked, so I guessed that one would have to add the element you want to change to the page that jasmine loads in the browser.

> Can we somewhere read anything about that spec.content construction?

https://github.com/diaspora/diaspora/blob/develop/spec/javascripts/helpers/SpecHelper.js#L149

zaziemo replied:
Thx for explaining!

denschub commented:
@Realtin 
> But we have to contact them.

Have you already asked them?

Realtin commented:
@denschub no, not yet. we wanted to ask if you would do it :)  but I know you have a lot of stuff on your plate. we can also do it today.

Realtin commented:
@denschub we'll write them now and put you in cc!

diaspora-code-review commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,47 @@
+          var map = L.map(mapContainer[0]).setView([location.lat, location.lng], 14);
+          L.tileLayer(tileLayerSource, {
+            id: mapsource.id,
+            accessToken: mapsource.access_token,
```
Identifier 'access_token' is not in camel case.

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -23,8 +26,52 @@ app.views.SinglePostContent = app.views.Base.extend({
+      var map = L.map(mapContainer[0]).setView([location.lat, location.lng], 14);
+      L.tileLayer(tileLayerSource, {
+        id: mapsource.id,
+        accessToken: mapsource.access_token,
```
Identifier 'access_token' is not in camel case.

diaspora-code-review commented on spec/javascripts/app/views/location_stream_spec.js:
```diff
@@ -0,0 +1,44 @@
+      it("should not initialize the map", function() {
+        expect(spec.content()).not.toContainElement(".mapContainer");
+      });
+    });
```
Expected an assignment or function call and instead saw an expression.

diaspora-code-review commented on spec/javascripts/app/views/single_post_content_view_spec.js:
```diff
@@ -0,0 +1,50 @@
+      it("should not initialize the map", function() {
+        expect(spec.content()).not.toContainElement(".mapContainer");
+      });
+    });
```
Expected an assignment or function call and instead saw an expression.

diaspora-code-review commented on spec/javascripts/app/views/location_stream_spec.js:
```diff
@@ -0,0 +1,43 @@
+  beforeEach(function(){
+    this.post = factory.post();
+    this.view = new app.views.LocationStream({model : this.post});
+    gon.appConfig = { map: {enabled: true, mapbox: {enabled: true, id: "yourID", access_token: "yourAccessToken" }} };
```
Identifier 'access_token' is not in camel case.

diaspora-code-review commented on spec/models/reshare_spec.rb:
```diff
@@ -244,57 +237,79 @@
+        end
+      end
+    end
+  end
```
`end` at 315, 2 is not aligned with `describe Reshare, type: :model do` at 3, 0

diaspora-code-review commented on spec/models/reshare_spec.rb:
```diff
@@ -244,57 +237,79 @@
+          expect(reshare.post_location[:lng]).to be_nil
+        end
+      end
+    end
```
`end` at 314, 4 is not aligned with `describe "#post_location" do` at 294, 2

diaspora-code-review commented on spec/models/reshare_spec.rb:
```diff
@@ -244,57 +237,79 @@
-end
+
+  describe "#post_location" do
+      let(:status_message) { build(:status_message, text: "This is a status_message", author: bob.person, public: true) }
```
Line is too long. [121/120]

diaspora-code-review commented on spec/models/reshare_spec.rb:
```diff
@@ -1,33 +1,33 @@
-describe Reshare, :type => :model do
-  it 'has a valid Factory' do
+describe Reshare, type: :model do
+  it "has a valid Factory" do
```
Use 2 (not 0) spaces for indentation.

jhass replied:
Uh, guess Rubocop got confused here?

jhass commented:
Got any response yet?

Realtin commented:
@jhass no nothing yet :(  
But we wrote a second reminder email today.

Realtin commented:
Good News Everyone!! We got the OK from University Heidelberg! 
We will make this PR ready today.

jhass commented:
That's great :+1:

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -1,7 +1,11 @@
+    "click .near-from": "toggleMap"
+  },
+
+  templateName: 'single-post-viewer/single-post-content',
```
Strings must use doublequote.

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -10,8 +14,7 @@ app.views.SinglePostContent = app.views.Base.extend({
     ".opengraph" : "openGraphView",
-    ".status-message-location" : "postLocationStreamView",
-    ".poll": "pollView"
+    '.poll': 'pollView',
```
Strings must use doublequote.

zaziemo committed:
add correct attributions for map tiles of Heidelberg University

remove personal credentials from mapbox and refactored some code
#5813

Signed-off-by: zaziemo <maren.heltsche@gmail.com>

diaspora-code-review commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,49 @@
+
+        if (location.lat) {
+
+          // If the mapbox option is enabled in the diaspora.yml, the mapbox tiles with the podmin's credentials are used.
```
Line is too long.

diaspora-code-review commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,49 @@
+        if (location.lat) {
+
+          // If the mapbox option is enabled in the diaspora.yml, the mapbox tiles with the podmin's credentials are used.
+          // If mapbox is not enabled the Maptiles from the Heidelberg University are used, which don't need credentials.
```
Line is too long.

diaspora-code-review commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,49 @@
+          // If the mapbox option is enabled in the diaspora.yml, the mapbox tiles with the podmin's credentials are used.
+          // If mapbox is not enabled the Maptiles from the Heidelberg University are used, which don't need credentials.
+          var mapsource = gon.appConfig.map.mapbox.enabled ? gon.appConfig.map.mapbox : "";
+          var tileLayerSource = mapsource ? "https://api.tiles.mapbox.com/v4/{id}/{z}/{x}/{y}.png?access_token={accessToken}"
```
Line is too long.

diaspora-code-review commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,49 @@
+          var attribution = "Map data &copy; <a href='http://openstreetmap.org'>OpenStreetMap</a> contributors, " +
+                            tileAttribution;
+
+          var map = L.map(mapContainer[0]).setView([location.lat, location.lng], 14);
```
'L' is not defined.

diaspora-code-review commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,49 @@
+                            tileAttribution;
+
+          var map = L.map(mapContainer[0]).setView([location.lat, location.lng], 14);
+          L.tileLayer(tileLayerSource, {
```
'L' is not defined.

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -23,8 +26,53 @@ app.views.SinglePostContent = app.views.Base.extend({
+      // If the mapbox option is enabled in the diaspora.yml, the mapbox tiles with the podmin's credentials are used.
+      // If mapbox is not enabled the Maptiles from the Heidelberg University are used, which don't need credentials.
+      var mapsource = gon.appConfig.map.mapbox.enabled ? gon.appConfig.map.mapbox : "";
+      var tileLayerSource = mapsource ? "https://api.tiles.mapbox.com/v4/{id}/{z}/{x}/{y}.png?access_token={accessToken}"
```
Line is too long.

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -23,8 +26,53 @@ app.views.SinglePostContent = app.views.Base.extend({
+      var attribution = "Map data &copy; <a href='http://openstreetmap.org'>OpenStreetMap</a> contributors, " +
+                        tileAttribution;
+
+      var map = L.map(mapContainer[0]).setView([location.lat, location.lng], 14);
```
'L' is not defined.

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -23,8 +26,53 @@ app.views.SinglePostContent = app.views.Base.extend({
+                        tileAttribution;
+
+      var map = L.map(mapContainer[0]).setView([location.lat, location.lng], 14);
+      L.tileLayer(tileLayerSource, {
```
'L' is not defined.

zaziemo commented:
@jhass do you mean we should revert the logic of the expression? Or create a whole new if/else logic?

jhass commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,49 @@
+
+          var map = L.map(mapContainer[0]).setView([location.lat, location.lng], 14);
+          L.tileLayer(tileLayerSource, {
+            id: mapsource.id,
```
Here you call `mapsource.id`. What's `"".id` supposed to return?

So yes, a `if/else` around everything different here would probably a bit cleaner and allow for example to not pass these attributes in the first place if they're not needed instead of passing undefined.

jhass commented on config/diaspora.yml.example:
```diff
@@ -324,6 +324,22 @@ configuration: ## Section
+  ## more reliable. There you have to create a account to get an ID and an access token which is limited. If
+  ## you want to get an unlimited account you can write an email
+  ## to team@diasporafoundation.org. Please enable mapbox and fill out your id
+  ## and access_token.
```
@denschub that offer still holds true?

jhass commented on config/diaspora.yml.example:
```diff
@@ -324,6 +324,22 @@ configuration: ## Section
 
+  ## Displays the location of a post in a map. Per default we are using the map
+  ## tiles of OpenMapSurfer (http://korona.geog.uni-heidelberg.de/).
+  ## If you want to use them you only have to enable the map. You also have the
```
I think this is either misleading or not what we want, that is we don't want a config option to turn off the feature completely, just one to switch from the default to mapbox.

zaziemo committed:
refactor code that choses the tile server based on podmin's choice

and remove the possibility to disable the map feature.
By default the application uses the itles of Heidelberg University that
don't need any credentials. If podmins enable the mapbox option in the
diaspora.yml and enter their credentials the mapbox tiles are used for the map
rendering.

diaspora-code-review commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,52 @@
+
+          tiles = L.tileLayer("https://api.tiles.mapbox.com/v4/{id}/{z}/{x}/{y}.png?access_token={accessToken}", {
+            id: gon.appConfig.map.mapbox.id,
+            accessToken: gon.appConfig.map.mapbox.access_token,
```
Identifier 'access_token' is not in camel case.

diaspora-code-review commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,52 @@
+      if (location.lat) {
+        // If map function is enabled the maptiles from the Heidelberg University are used by default.
+
+        var map = L.map(mapContainer[0]).setView([location.lat, location.lng], 14);
```
'L' is not defined.

diaspora-code-review commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,52 @@
+
+        var map = L.map(mapContainer[0]).setView([location.lat, location.lng], 14);
+
+        var tiles = L.tileLayer("http://korona.geog.uni-heidelberg.de/tiles/roads/x={x}&y={y}&z={z}", {
```
'L' is not defined.

diaspora-code-review commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,52 @@
+
+        tiles.addTo(map);
+
+        var markerOnMap = L.marker(location).addTo(map);
```
'L' is not defined.

diaspora-code-review replied:
'markerOnMap' is defined but never used.

svbergerem replied:
Since markerOnMap seems to be unused you could remove it:
```js
L.marker(location).addTo(map);
```

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -23,8 +26,57 @@ app.views.SinglePostContent = app.views.Base.extend({
+
+        tiles = L.tileLayer("https://api.tiles.mapbox.com/v4/{id}/{z}/{x}/{y}.png?access_token={accessToken}", {
+          id: gon.appConfig.map.mapbox.id,
+          accessToken: gon.appConfig.map.mapbox.access_token,
```
Identifier 'access_token' is not in camel case.

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -23,8 +26,57 @@ app.views.SinglePostContent = app.views.Base.extend({
+
+      // If map function is enabled the maptiles from the Heidelberg University are used by default.
+
+      var map = L.map(mapContainer[0]).setView([location.lat, location.lng], 14);
```
'L' is not defined.

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -23,8 +26,57 @@ app.views.SinglePostContent = app.views.Base.extend({
+
+      var map = L.map(mapContainer[0]).setView([location.lat, location.lng], 14);
+
+      var tiles = L.tileLayer("http://korona.geog.uni-heidelberg.de/tiles/roads/x={x}&y={y}&z={z}", {
```
'L' is not defined.

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -23,8 +26,57 @@ app.views.SinglePostContent = app.views.Base.extend({
+      // If the mapbox option is enabled in the diaspora.yml, the mapbox tiles with the podmin's credentials are used.
+      if (gon.appConfig.map.mapbox.enabled) {
+
+        tiles = L.tileLayer("https://api.tiles.mapbox.com/v4/{id}/{z}/{x}/{y}.png?access_token={accessToken}", {
```
'L' is not defined.

zaziemo commented:
we think it is ready to merge now! hopefully ;)

svbergerem commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,52 @@
+                        "rendering <a href='http://giscience.uni-hd.de/'>" +
+                        "GIScience Research Group @ Heidelberg University</a>",
+          maxZoom: 18,
+          });
```
Maybe remove 2 spaces here?
```js
var tiles = L.tileLayer(..., {
  attribution: ...,
  maxZoom: 18
});
```

svbergerem commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,52 @@
+      }
+    } else {
+        mapContainer.toggle();
+      }
```
The indentation looks weird.
```js
if (...) {
} else {
  mapContainer.toggle();
}
```

svbergerem commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -23,8 +26,57 @@ app.views.SinglePostContent = app.views.Base.extend({
+                       "Imagery © <a href='https://www.mapbox.com'>Mapbox</a>",
+          maxZoom: 18,
+        });
+      }
```
This could be a helper function (`getTiles`) since you seem to have the same code in `app/assets/javascripts/app/views/location_stream.js`. (You don't have to do it in this PR).

svbergerem commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -23,8 +26,57 @@ app.views.SinglePostContent = app.views.Base.extend({
+  toggleMap: function () {
+      $(".mapContainer").height($(".small-map")[0] ? 200 : 50);
+      $(".leaflet-control-zoom").css("display", $(".small-map")[0] ? "block" : "none");
+      $(".mapContainer").toggleClass("small-map");
```
Use 2 whitespaces for indentation.

svbergerem commented on config/.jshint.json:
```diff
@@ -53,7 +53,7 @@
     "qq",
     "blueimp",
-
+    "L",
```
We have 3 blocks here:

1. defined by our gems
2. for the jasmine tests
3. our own global variables

Please put the new variable in the first block (ordered alphabetically) and keep the empty line.

svbergerem commented on spec/javascripts/app/views/location_stream_spec.js:
```diff
@@ -0,0 +1,43 @@
+        this.view.toggleMap();
+        expect($(".mapContainer")).toHaveCss({display: "none"});
+      });
+    }),
```
`});`

svbergerem commented on spec/javascripts/app/views/single_post_content_view_spec.js:
```diff
@@ -0,0 +1,50 @@
+        this.view.toggleMap();
+        expect($(".mapContainer").height() < 100).toBeTruthy();
+      });
+    }),
```
`});`

Realtin committed:
fix indentation & jshint predefines

according to svbergerem's annotations

Realtin commented:
@svbergerem thank you! ✨

jhass commented on spec/models/status_message_spec.rb:
```diff
@@ -63,48 +61,45 @@
-      sm2 = FactoryGirl.create(:status_message, :author => bob.person)
+    it "returns an array of the status_message guids" do
+      status_message_1 = FactoryGirl.create(:status_message, author: alice.person)
+      status_message_2 = FactoryGirl.create(:status_message, author: bob.person)
```
Ping

jhass commented on spec/models/status_message_spec.rb:
```diff
@@ -233,222 +227,240 @@
-    it 'returns MatchObject (true) if the post contains #nsfw (however capitalised)' do
-      status  = FactoryGirl.build(:status_message, :text => "This message is #nSFw")
+    it "returns MatchObject (true) if the post contains #nsfw (however capitalised)" do
+      status  = FactoryGirl.build(:status_message, text: "This message is #nSFw")
```
Ping

jhass commented on spec/models/status_message_spec.rb:
```diff
@@ -233,222 +227,240 @@
-    it 'returns nil (false) if the post does not contain #nsfw' do
-      status  = FactoryGirl.build(:status_message, :text => "This message is #sFW")
+    it "returns nil (false) if the post does not contain #nsfw" do
+      status  = FactoryGirl.build(:status_message, text: "This message is #sFW")
```
Ping

svbergerem commented:
Pronto still finds some codestyle issues:
```
app/assets/javascripts/app/views/location_stream.js:33 W: Identifier 'access_token' is not in camel case.
app/assets/javascripts/app/views/location_stream.js:43 W: 'markerOnMap' is defined but never used.
app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:56 W: Identifier 'access_token' is not in camel case.
app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:71 W: 'markerOnMap' is defined but never used.
spec/javascripts/app/views/location_stream_spec.js:5 W: Identifier 'access_token' is not in camel case.
spec/javascripts/app/views/locator_spec.js:3 W: Read only.
spec/javascripts/helpers/factory.js:153 W: Line is too long.
spec/factories.rb:146 W: Ambiguous negative number operator. Parenthesize the method arguments if it's surely a negative number operator, or add a whitespace to the right of the `-` if it should be a subtraction.
spec/models/status_message_spec.rb:66 W: Useless assignment to variable - `status_message_2`. Did you mean `status_message_1`?
```

For the camel case issues you can add surrounding comments to disable the camel case check and enable it afterwards: https://github.com/diaspora/diaspora/blob/develop/spec/javascripts/app/models/pod_spec.js#L9

`spec/javascripts/app/views/locator_spec.js:3 W: Read only.`
Using `window.OSM` instead of `OSM` should fix this.

While testing this PR on my dev pod I also found two UX issues:

1. In the stream view the map doesn't display the plus and minus buttons to zoom in and out.
2. In the SPV I don't think that it is obvious that you have to click on the location to expand the map which is already displayed. Maybe display the map already expanded on load and remove the toggle function for the SPV?

Flaburgan commented:
>Maybe display the map already expanded on load and remove the toggle function for the SPV?

+1 on this.

Realtin committed:
fix all the pronto remarks finally

- fix map controls in stream

diaspora-code-review commented on app/assets/javascripts/app/views/location_stream.js:
```diff
@@ -1,7 +1,54 @@
+
+        tiles.addTo(map);
+
+        L.marker(location).addTo(map);
```
'L' is not defined.

Realtin replied:
local meckert der aber nicht !!!1111!1!11!1 :)

jhass replied:
Yes, you can ignore these.

diaspora-code-review commented on app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js:
```diff
@@ -23,8 +26,59 @@ app.views.SinglePostContent = app.views.Base.extend({
+      map.invalidateSize();
+
+      // put marker on map
+      L.marker(location).addTo(map);
```
'L' is not defined.

zaziemo commented:
@Flaburgan wasn't it you who suggested to have a small map displayed by default and an expanded view by clicking? ;)

Realtin commented:
Ok so I fixed all the pronto issues or so I thought..  at least localy it doesn't have any complaints anymore. 
The map controls are also back in the stream view map (this could probably be done more elegantly) 

as for the UI/UX Issues.. we thought about these a lot. we experimented with icons and text indicating the map and so on but couldn't really find a good solution or the other solutions were dismissed in this pull request discussion. Maybe we could do this in a later PR?

Flaburgan commented:
@zaziemo hm maybe I misunderstood Steffen's comment then :p

svbergerem commented:
> Maybe we could do this in a later PR? 

Yes, that would be ok for me. I think this PR is now **ready to merge**. When we merge this I'd like to have two new issues:

1. Refactor getting tiles: https://github.com/diaspora/diaspora/pull/6256#discussion_r39148762
2. „In the SPV I don't think that it is obvious that you have to click on the location to expand the map which is already displayed. Maybe display the map already expanded on load and remove the toggle function for the SPV?“

svbergerem commented:
Thanks a lot! We decided to merge it into the develop branch because there were too many conflicts on the stable branch.

denschub commented:
Thank you very much to everyone who was involved! Good work! :)

zaziemo commented:
thank you all for the great feedback and help! It was fun working on this :)
